src/coreclr/jit/emit.h

```diff
@@ -2180,6 +2180,13 @@ class emitter
 
     instrDesc* emitLastIns;
 
+    // Check if a peephole opt involving emitLastIns is unsafe
+    // emitForceNewIG here prevents peepholes from crossing nogc boundaries.
+    bool emitCannotPeepholeLastIns()
+    {
+        return emitForceNewIG || ((emitCurIGinsCnt == 0) && ((emitCurIG->igFlags & IGF_EXTEND) == 0));
+    }
```

Comment:
I'm not fond of functions with "Not" in the name, especially when they are later used with `!`. How about `emitPeepholeLastInsIsSafe`? How about also checking `emitLastIns != nullptr` since that is also checked at most of the call sites.

```suggestion
    bool emitPeepholeLastInsIsSafe()
    {
        return (emitLastIns != nullptr) && !emitForceNewIG && ((emitCurIGinsCnt != 0) || ((emitCurIG->igFlags & IGF_EXTEND) != 0));
    }
```

Reply:
Done.